diff --git a/.gitignore b/.gitignore
index 451520d1e48..2e0b2ed13c2 100644
--- a/.gitignore
+++ b/.gitignore
@@ -63,4 +63,6 @@ unifyconfig/unifyproject.config
 customerapp/src/main/assets/df.cfg
 sellerapp/src/main/assets/df.cfg
 
-jacoco.exec
\ No newline at end of file
+jacoco.exec
+/robust/
+/robust/
\ No newline at end of file
diff --git a/build.gradle b/build.gradle
index 6d78b6578dd..8b9847a7204 100644
--- a/build.gradle
+++ b/build.gradle
@@ -24,15 +24,19 @@ buildscript {
                 username = "${artifactory_username}"
                 password = "${artifactory_password}"
             }
+            allowInsecureProtocol = true
         }
         maven { url 'https://plugins.gradle.org/m2/' }
     }
     dependencies {
-        classpath ('com.android.tools.build:gradle:4.2.1') {
+        classpath ('com.android.tools.build:gradle:7.0.4') {
             exclude module: 'proguard-gradle'
         }
+//        classpath ('com.android.tools.build:gradle:4.2.1') {
+//            exclude module: 'proguard-gradle'
+//        }
         classpath "org.jacoco:org.jacoco.core:$jacocoVersion"
-        classpath('net.sf.proguard:proguard-gradle:6.0') {
+        classpath('net.sf.proguard:proguard-gradle:6.0.3') {
             force = true
         }
         classpath("org.jfrog.buildinfo:build-info-extractor-gradle:4.18.2") {
@@ -48,7 +52,7 @@ buildscript {
         classpath 'com.tokopedia.patch:core:1.0.0'
         classpath 'org.javassist:javassist:3.20.0-GA'
         classpath "androidx.benchmark:benchmark-gradle-plugin:1.0.0"
-        classpath "android.arch.navigation:navigation-safe-args-gradle-plugin:1.0.0"
+        classpath "androidx.navigation:navigation-safe-args-gradle-plugin:2.3.5"
         classpath "com.newrelic.agent.android:agent-gradle-plugin:$newRelicVersion"
         classpath "com.gradle:gradle-enterprise-gradle-plugin:3.6.3"
         classpath "gradle.plugin.org.gradle.android:android-cache-fix-gradle-plugin:2.4.4"
@@ -66,11 +70,11 @@ apply from: 'buildproj.gradle'
 
 apply from: "$rootDir/tools/set_debug_tools.gradle"
 
-subprojects{
-    plugins.withType(com.android.build.gradle.BasePlugin) {
-        project.apply plugin: "org.gradle.android.cache-fix"
-    }
-}
+//subprojects{
+//    plugins.withType(com.android.build.gradle.BasePlugin) {
+//        project.apply plugin: "org.gradle.android.cache-fix"
+//    }
+//}
 
 //Installing Git Hooks
 task installGitHooks {
diff --git a/buildconfig/dependencies/dependencies.gradle b/buildconfig/dependencies/dependencies.gradle
index 4ed751ea1eb..b80537bc84e 100755
--- a/buildconfig/dependencies/dependencies.gradle
+++ b/buildconfig/dependencies/dependencies.gradle
@@ -1,6 +1,6 @@
 ext {
 
-    compileSdkVersion = 30
+    compileSdkVersion = 31
     minSdkVersion = 21
     targetSdkVersion = 30
     buildToolsVersion = "30.0.2"
@@ -8,7 +8,7 @@ ext {
     jacocoVersion = "0.8.7"
     kotlinVersion = "1.5.30"
 
-    navigationVersion = "2.3.0"
+    navigationVersion = "2.3.5"
     lifeCycleVersion = "2.2.0"
     lifeCycleExtensionsVersion = "2.2.0"
     archCoreVersion = "2.1.0"
@@ -39,7 +39,7 @@ ext {
     apolloVersion = '0.3.0'
     lottieVersion = '3.4.1'
     cardioVersion = '5.5.1'
-    roomVersion = '2.3.0'
+    roomVersion = '2.4.1'
     sqliteFrameworkVersion = '2.1.0'
     multiDexVersion = '2.0.0'
     twitter4JVersion = '4.0.7'
@@ -78,7 +78,7 @@ ext {
     kotlinTestVersion = '1.5.30'
     KotlinMockitoVersion = '1.5.0'
     SpekVersion = "1.1.5"
-    crashlyticsGradleVersion = '2.2.0'
+    crashlyticsGradleVersion = '2.5.2'
     googleServiceGradleVersion = '4.3.3'
     chuckerVersion = '4.0.6-tkpd'
     sliceVersion = '1.0.0-alpha6'
@@ -249,7 +249,7 @@ ext {
             chips                 : 'com.beloo.widget:ChipsLayoutManager:0.3.7@aar',
             roundCornerProgressBar: 'com.akexorcist:RoundCornerProgressBar:2.0.3',
             lottie                : "com.airbnb.android:lottie:$lottieVersion",
-            hansel                : "io.hansel.preprocessor:preprocessor:9.1.5",
+            hansel                : "io.hansel.preprocessor:preprocessor:9.2.0",
             protobuf              : "com.google.protobuf:protobuf-gradle-plugin:0.8.12",
             autopatchbase         : "com.tokopedia.patch:core:1.0.0",
             apacheLang            : "org.apache.commons:commons-lang3:3.0",
diff --git a/buildconfig/kotlin/core-module.gradle b/buildconfig/kotlin/core-module.gradle
index 8911910e8e5..6c1565f7c27 100644
--- a/buildconfig/kotlin/core-module.gradle
+++ b/buildconfig/kotlin/core-module.gradle
@@ -25,12 +25,12 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8.toString()
+        jvmTarget = JavaVersion.VERSION_11.toString()
     }
 
     testOptions {
diff --git a/buildconfig/kotlin/module.gradle b/buildconfig/kotlin/module.gradle
index 91ddcd4edaf..609fe30383b 100644
--- a/buildconfig/kotlin/module.gradle
+++ b/buildconfig/kotlin/module.gradle
@@ -42,12 +42,12 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8.toString()
+        jvmTarget = JavaVersion.VERSION_11.toString()
     }
 
     testOptions {
diff --git a/buildconfig/module.gradle b/buildconfig/module.gradle
index 46dd337024a..e7f30162756 100644
--- a/buildconfig/module.gradle
+++ b/buildconfig/module.gradle
@@ -1,5 +1,5 @@
 def isDebuggable() {
-    if (project.hasProperty('enableDebug') && enableDebug == "true") {
+    if (rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true") {
         return "true";
     }
     return "false";
@@ -48,8 +48,8 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     testOptions {
diff --git a/buildproj.gradle b/buildproj.gradle
index 94dfbffe80c..22301700db2 100644
--- a/buildproj.gradle
+++ b/buildproj.gradle
@@ -13,6 +13,7 @@ allprojects {
                 username = "${artifactory_username}"
                 password = "${artifactory_password}"
             }
+            allowInsecureProtocol = true
         }
         maven {
             url "https://jitpack.io"
diff --git a/customer_mid_app/build.gradle b/customer_mid_app/build.gradle
index 4a110067f7d..9c893be797d 100644
--- a/customer_mid_app/build.gradle
+++ b/customer_mid_app/build.gradle
@@ -29,8 +29,8 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
     packagingOptions {
         exclude 'META-INF/**'
@@ -102,6 +102,7 @@ dependencies {
     implementation projectOrAar(rootProject.ext.libraries.collapsingTabLayout)
     implementation projectOrAar(rootProject.ext.libraries.videoplayer)
     implementation projectOrAar(rootProject.ext.libraries.websocket)
+    implementation projectOrAar(rootProject.ext.libraries.tokopatch)
     implementation projectOrAar(rootProject.ext.libraries.baseAdapter)
     implementation projectOrAar(rootProject.ext.features.csat_rating)
     implementation projectOrAar(rootProject.ext.features.resourcesCommon)
diff --git a/customer_mid_app/src/main/java/com/tokopedia/tkpd/app/ConsumerMainApplication.java b/customer_mid_app/src/main/java/com/tokopedia/tkpd/app/ConsumerMainApplication.java
index 76af688218d..0913db6b9b7 100644
--- a/customer_mid_app/src/main/java/com/tokopedia/tkpd/app/ConsumerMainApplication.java
+++ b/customer_mid_app/src/main/java/com/tokopedia/tkpd/app/ConsumerMainApplication.java
@@ -82,6 +82,7 @@ import com.tokopedia.unifyprinciples.Typography;
 import com.tokopedia.url.TokopediaUrl;
 import com.tokopedia.weaver.WeaveInterface;
 import com.tokopedia.weaver.Weaver;
+import com.tokopedia.tokopatch.TokoPatch;
 
 import org.jetbrains.annotations.NotNull;
 
@@ -147,6 +148,7 @@ public abstract class ConsumerMainApplication extends ConsumerRouterApplication
         com.tokopedia.akamai_bot_lib.UtilsKt.initAkamaiBotManager(ConsumerMainApplication.this);
         createAndCallPreSeq();
         super.onCreate();
+        TokoPatch.init(this);
         createAndCallPostSeq();
         initializeAbTestVariant();
         createAndCallFetchAbTest();
diff --git a/customerapp/build.gradle b/customerapp/build.gradle
index e630191ba73..15d91772248 100644
--- a/customerapp/build.gradle
+++ b/customerapp/build.gradle
@@ -3,6 +3,7 @@ apply plugin: 'embrace-swazzler'
 apply plugin: 'com.google.firebase.crashlytics'
 apply plugin: 'kotlin-android'
 apply plugin: 'newrelic'
+apply plugin: 'robust'
 
 if (project.hasProperty("enable-shrinkresource")) {
     apply from: "$rootProject.projectDir/buildconfig/appcompile/enable-shrinkresource.gradle"
@@ -177,7 +178,7 @@ android {
             if (devMode()) {
                 signingConfig signingConfigs.debug
             }
-            debuggable = project.hasProperty('enableDebug') && enableDebug == "true"
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
             ext.enableHansel = !project.hasProperty('hanselDisableRelease')
 
             if (isFirebasePerformanceLogEnabled().toBoolean()) {
@@ -228,12 +229,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8.toString()
+        jvmTarget = JavaVersion.VERSION_11.toString()
     }
 
     configurations.all {
diff --git a/customerapp/proguard-rules.txt b/customerapp/proguard-rules.txt
index fdbaafb2bed..36f67f923ca 100644
--- a/customerapp/proguard-rules.txt
+++ b/customerapp/proguard-rules.txt
@@ -1,3 +1,7 @@
+-printusage usage.txt
+-printseeds seeds.txt
+-printconfiguration config.txt
+
 # Obfuscation parameters:
 -dontwarn java.nio.file.Files
 -dontwarn java.nio.file.Path
@@ -20,7 +24,7 @@
 -dontwarn com.squareup.okhttp.**
 -dontwarn okio.**
 -dontwarn com.viewpagerindicator.LinePageIndicator
--dontwarn sun.misc.Unsafe
+-dontwarn sun.misc.**
 
 -keep class id.co.veritrans.android.api.**{ *;}
 -keepclassmembers class id.co.veritrans.android.api.**{
@@ -38,11 +42,12 @@
 -keep public class com.google.android.gms.common.internal.safeparcel.SafeParcelable {
     public static final *** NULL;
 }
--keepnames @com.google.android.gms.common.annotation.KeepName class *
+-keep @com.google.android.gms.common.annotation.KeepName class *
 -keepclassmembernames class * {
     @com.google.android.gms.common.annotation.KeepName *;
 }
--keepnames class * implements android.os.Parcelable {
+
+-keep class * implements android.os.Parcelable {
     public static final ** CREATOR;
 }
 #end com.google.android.gms
@@ -162,8 +167,8 @@
 -dontwarn kotlinx.coroutines.flow.**
 -dontwarn kotlinx.coroutines.reactive.**
 
--keepnames class kotlinx.coroutines.internal.MainDispatcherFactory {}
--keepnames class kotlinx.coroutines.CoroutineExceptionHandler {}
+-keep class kotlinx.coroutines.internal.MainDispatcherFactory {}
+-keep class kotlinx.coroutines.CoroutineExceptionHandler {}
 
 # -- App indexing
 -dontwarn com.google.firebase.appindexing.**
@@ -243,15 +248,15 @@
 
 
 # To show pop-up via CM engine
--keepnames class com.tokopedia.home.beranda.presentation.view.fragment.HomeRevampFragment
--keepnames class com.tokopedia.officialstore.category.presentation.fragment.OfficialHomeContainerFragment
--keepnames class com.tokopedia.home_recom.view.fragment.RecommendationFragment
--keepnames class com.tokopedia.home_recom.view.fragment.SimilarProductRecommendationFragment
--keepnames class com.tokopedia.feedplus.view.fragment.FeedPlusContainerFragment
--keepnames class com.tokopedia.cart.view.CartFragment
--keepnames class com.tokopedia.dynamicfeatures.DFInstallerFragment
--keepnames class com.tokopedia.review.feature.reading.presentation.fragment.ShopReviewFragment
--keepnames class com.tokopedia.shop_showcase.shop_showcase_tab.presentation.fragment.ShopPageShowcaseFragment
+-keep class com.tokopedia.home.beranda.presentation.view.fragment.HomeRevampFragment
+-keep class com.tokopedia.officialstore.category.presentation.fragment.OfficialHomeContainerFragment
+-keep class com.tokopedia.home_recom.view.fragment.RecommendationFragment
+-keep class com.tokopedia.home_recom.view.fragment.SimilarProductRecommendationFragment
+-keep class com.tokopedia.feedplus.view.fragment.FeedPlusContainerFragment
+-keep class com.tokopedia.cart.view.CartFragment
+-keep class com.tokopedia.dynamicfeatures.DFInstallerFragment
+-keep class com.tokopedia.review.feature.reading.presentation.fragment.ShopReviewFragment
+-keep class com.tokopedia.shop_showcase.shop_showcase_tab.presentation.fragment.ShopPageShowcaseFragment
 
 # feed_shop
 -keepnames class com.tokopedia.feed_shop.shop.view.fragment.FeedShopFragment
@@ -288,4 +293,21 @@
 -keepattributes Exceptions, Signature, InnerClasses, LineNumberTable, SourceFile, EnclosingMethod
 
 # Window
--keep class androidx.window.** { *; }
\ No newline at end of file
+-keep class androidx.window.** { *; }
+
+-keep class io.jsonwebtoken.** { *; }
+-keepnames class io.jsonwebtoken.* { *; }
+-keepnames interface io.jsonwebtoken.* { *; }
+
+-keep class org.bouncycastle.** { *; }
+-keepnames class org.bouncycastle.** { *; }
+-dontwarn org.bouncycastle.**
+
+-keep class com.tokopedia.stability.**{*;}
+-keepclassmembers class **{
+public static com.tokopedia.stability.ChangeDelegate *;
+}
+
+-keepclassmembers,allowobfuscation class * {
+ @com.google.gson.annotations.SerializedName <fields>;
+}
\ No newline at end of file
diff --git a/customerapp_pro/build.gradle b/customerapp_pro/build.gradle
index cda78099923..052f8bbc8c2 100644
--- a/customerapp_pro/build.gradle
+++ b/customerapp_pro/build.gradle
@@ -142,7 +142,7 @@ android {
         htmlReport false
         warningsAsErrors false
         abortOnError false
-        check lintchecks
+        checkOnly lintchecks
     }
 
     signingConfigs {
@@ -177,7 +177,7 @@ android {
             if (devMode()) {
                 signingConfig signingConfigs.debug
             }
-            debuggable = project.hasProperty('enableDebug') && enableDebug == "true"
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
             ext.enableHansel = !project.hasProperty('hanselDisableRelease')
 
             if (isFirebasePerformanceLogEnabled().toBoolean()) {
@@ -230,8 +230,8 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     configurations.all {
diff --git a/customerapp_pro/proguard-rules.txt b/customerapp_pro/proguard-rules.txt
index 38d7c87642b..e7dd414d290 100644
--- a/customerapp_pro/proguard-rules.txt
+++ b/customerapp_pro/proguard-rules.txt
@@ -1,3 +1,7 @@
+-printusage usage.txt
+-printseeds seeds.txt
+-printconfiguration config.txt
+
 # Obfuscation parameters:
 -dontwarn java.nio.file.Files
 -dontwarn java.nio.file.Path
@@ -38,11 +42,12 @@
 -keep public class com.google.android.gms.common.internal.safeparcel.SafeParcelable {
     public static final *** NULL;
 }
--keepnames @com.google.android.gms.common.annotation.KeepName class *
+-keep @com.google.android.gms.common.annotation.KeepName class *
 -keepclassmembernames class * {
     @com.google.android.gms.common.annotation.KeepName *;
 }
--keepnames class * implements android.os.Parcelable {
+
+-keep class * implements android.os.Parcelable {
     public static final ** CREATOR;
 }
 #end com.google.android.gms
@@ -161,8 +166,8 @@
 -dontwarn kotlinx.coroutines.flow.**inlined**
 -dontwarn kotlinx.coroutines.flow.**
 
--keepnames class kotlinx.coroutines.internal.MainDispatcherFactory {}
--keepnames class kotlinx.coroutines.CoroutineExceptionHandler {}
+-keep class kotlinx.coroutines.internal.MainDispatcherFactory {}
+-keep class kotlinx.coroutines.CoroutineExceptionHandler {}
 
 # -- App indexing
 -dontwarn com.google.firebase.appindexing.**
@@ -235,16 +240,16 @@
 -dontwarn com.tokopedia.sortfilter.SortFilter*
 
 # To show pop-up via CM engine
--keepnames class com.tokopedia.home.beranda.presentation.view.fragment.HomeFragment
--keepnames class com.tokopedia.home.beranda.presentation.view.fragment.HomeRevampFragment
--keepnames class com.tokopedia.officialstore.category.presentation.fragment.OfficialHomeContainerFragment
--keepnames class com.tokopedia.home_recom.view.fragment.RecommendationFragment
--keepnames class com.tokopedia.home_recom.view.fragment.SimilarProductRecommendationFragment
--keepnames class com.tokopedia.feedplus.view.fragment.FeedPlusContainerFragment
--keepnames class com.tokopedia.cart.view.CartFragment
--keepnames class com.tokopedia.dynamicfeatures.DFInstallerFragment
--keepnames class com.tokopedia.review.feature.reading.presentation.fragment.ShopReviewFragment
--keepnames class com.tokopedia.shop_showcase.shop_showcase_tab.presentation.fragment.ShopPageShowcaseFragment
+-keep class com.tokopedia.home.beranda.presentation.view.fragment.HomeFragment
+-keep class com.tokopedia.home.beranda.presentation.view.fragment.HomeRevampFragment
+-keep class com.tokopedia.officialstore.category.presentation.fragment.OfficialHomeContainerFragment
+-keep class com.tokopedia.home_recom.view.fragment.RecommendationFragment
+-keep class com.tokopedia.home_recom.view.fragment.SimilarProductRecommendationFragment
+-keep class com.tokopedia.feedplus.view.fragment.FeedPlusContainerFragment
+-keep class com.tokopedia.cart.view.CartFragment
+-keep class com.tokopedia.dynamicfeatures.DFInstallerFragment
+-keep class com.tokopedia.review.feature.reading.presentation.fragment.ShopReviewFragment
+-keep class com.tokopedia.shop_showcase.shop_showcase_tab.presentation.fragment.ShopPageShowcaseFragment
 
 # feed_shop
 -keepnames class com.tokopedia.feed_shop.shop.view.fragment.FeedShopFragment
@@ -279,4 +284,17 @@
 # New Relic
 -keep class com.newrelic.** { *; }
 -dontwarn com.newrelic.**
--keepattributes Exceptions, Signature, InnerClasses, LineNumberTable, SourceFile, EnclosingMethod
\ No newline at end of file
+-keepattributes Exceptions, Signature, InnerClasses, LineNumberTable, SourceFile, EnclosingMethod
+
+-keep class com.tokopedia.stability.**{*;}
+-keepclassmembers class **{
+public static com.tokopedia.stability.ChangeDelegate *;
+}
+
+-keepclassmembers,allowobfuscation class * {
+ @com.google.gson.annotations.SerializedName <fields>;
+}
+
+-keep class com.google.gson.reflect.TypeToken { *; }
+-keep class * extends com.google.gson.reflect.TypeToken
+-keep class kotlin.coroutines.Continuation
\ No newline at end of file
diff --git a/dynamic-features-sellerapp/base/df_base_sellerapp/build.gradle b/dynamic-features-sellerapp/base/df_base_sellerapp/build.gradle
index 9684990dd3f..812f41d804b 100644
--- a/dynamic-features-sellerapp/base/df_base_sellerapp/build.gradle
+++ b/dynamic-features-sellerapp/base/df_base_sellerapp/build.gradle
@@ -36,6 +36,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features-sellerapp/merchant/shop_score_sellerapp/build.gradle b/dynamic-features-sellerapp/merchant/shop_score_sellerapp/build.gradle
index b55d3cbe156..4b9628bb9e1 100644
--- a/dynamic-features-sellerapp/merchant/shop_score_sellerapp/build.gradle
+++ b/dynamic-features-sellerapp/merchant/shop_score_sellerapp/build.gradle
@@ -29,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features-sellerapp/topads/seller_topads_dashboard/build.gradle b/dynamic-features-sellerapp/topads/seller_topads_dashboard/build.gradle
index 079de25b34a..e451760a437 100644
--- a/dynamic-features-sellerapp/topads/seller_topads_dashboard/build.gradle
+++ b/dynamic-features-sellerapp/topads/seller_topads_dashboard/build.gradle
@@ -28,6 +28,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/base/df_base/build.gradle b/dynamic-features/base/df_base/build.gradle
index 4aad9096e3c..5675e80bba8 100644
--- a/dynamic-features/base/df_base/build.gradle
+++ b/dynamic-features/base/df_base/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/category/df_category_trade_in/build.gradle b/dynamic-features/category/df_category_trade_in/build.gradle
index e30e5331148..b40cf40dc62 100644
--- a/dynamic-features/category/df_category_trade_in/build.gradle
+++ b/dynamic-features/category/df_category_trade_in/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -45,6 +46,12 @@ android {
             dimension "client"
         }
     }
+
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
 }
 
 dependencies {
diff --git a/dynamic-features/common/df_alpha_testing/build.gradle b/dynamic-features/common/df_alpha_testing/build.gradle
index fad66fcdd64..e1c8fe7ee2c 100644
--- a/dynamic-features/common/df_alpha_testing/build.gradle
+++ b/dynamic-features/common/df_alpha_testing/build.gradle
@@ -28,6 +28,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/content/df_content_play_broadcaster/build.gradle b/dynamic-features/content/df_content_play_broadcaster/build.gradle
index 87fe13dc41e..c2cdf788e73 100644
--- a/dynamic-features/content/df_content_play_broadcaster/build.gradle
+++ b/dynamic-features/content/df_content_play_broadcaster/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/content/df_createpost/build.gradle b/dynamic-features/content/df_createpost/build.gradle
index ef930a20d47..2bec055ff2a 100644
--- a/dynamic-features/content/df_createpost/build.gradle
+++ b/dynamic-features/content/df_createpost/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/content/df_imagepicker_insta/build.gradle b/dynamic-features/content/df_imagepicker_insta/build.gradle
index 9a5dd2ef6bb..cb7e2eb4d71 100644
--- a/dynamic-features/content/df_imagepicker_insta/build.gradle
+++ b/dynamic-features/content/df_imagepicker_insta/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/content/df_people/build.gradle b/dynamic-features/content/df_people/build.gradle
index 8d039c791be..e32814956c9 100644
--- a/dynamic-features/content/df_people/build.gradle
+++ b/dynamic-features/content/df_people/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/entertainment/df_entertainment/build.gradle b/dynamic-features/entertainment/df_entertainment/build.gradle
index e1a325a43b0..906f7338784 100644
--- a/dynamic-features/entertainment/df_entertainment/build.gradle
+++ b/dynamic-features/entertainment/df_entertainment/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/merchant/df_merchant_login/build.gradle b/dynamic-features/merchant/df_merchant_login/build.gradle
index 81c69a5b960..d99309c3aaf 100644
--- a/dynamic-features/merchant/df_merchant_login/build.gradle
+++ b/dynamic-features/merchant/df_merchant_login/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/merchant/df_merchant_nonlogin/build.gradle b/dynamic-features/merchant/df_merchant_nonlogin/build.gradle
index 700870c0e09..b07781ea5d6 100644
--- a/dynamic-features/merchant/df_merchant_nonlogin/build.gradle
+++ b/dynamic-features/merchant/df_merchant_nonlogin/build.gradle
@@ -28,6 +28,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/merchant/df_merchant_seller/build.gradle b/dynamic-features/merchant/df_merchant_seller/build.gradle
index 2afde79cba2..a41706c09b3 100644
--- a/dynamic-features/merchant/df_merchant_seller/build.gradle
+++ b/dynamic-features/merchant/df_merchant_seller/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/operational/df_operational_contact_us/build.gradle b/dynamic-features/operational/df_operational_contact_us/build.gradle
index 8163b76bef5..823e369742e 100644
--- a/dynamic-features/operational/df_operational_contact_us/build.gradle
+++ b/dynamic-features/operational/df_operational_contact_us/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/promo/df_promo_gamification/build.gradle b/dynamic-features/promo/df_promo_gamification/build.gradle
index a0cf69878e5..15a0c30edd1 100644
--- a/dynamic-features/promo/df_promo_gamification/build.gradle
+++ b/dynamic-features/promo/df_promo_gamification/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/promo/df_promo_tokopoints/build.gradle b/dynamic-features/promo/df_promo_tokopoints/build.gradle
index 933e1150ccd..e332c314b3c 100644
--- a/dynamic-features/promo/df_promo_tokopoints/build.gradle
+++ b/dynamic-features/promo/df_promo_tokopoints/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/recharge/df_recharge_brizzi/build.gradle b/dynamic-features/recharge/df_recharge_brizzi/build.gradle
index 92b312db62e..3a504d78ed5 100644
--- a/dynamic-features/recharge/df_recharge_brizzi/build.gradle
+++ b/dynamic-features/recharge/df_recharge_brizzi/build.gradle
@@ -28,6 +28,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/recharge/df_recharge_emoney/build.gradle b/dynamic-features/recharge/df_recharge_emoney/build.gradle
index 4d4e52ec837..5768e84181f 100644
--- a/dynamic-features/recharge/df_recharge_emoney/build.gradle
+++ b/dynamic-features/recharge/df_recharge_emoney/build.gradle
@@ -28,6 +28,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/salam/df_salam_umrah/build.gradle b/dynamic-features/salam/df_salam_umrah/build.gradle
index 47209140bb9..357374b726e 100644
--- a/dynamic-features/salam/df_salam_umrah/build.gradle
+++ b/dynamic-features/salam/df_salam_umrah/build.gradle
@@ -28,6 +28,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/tokopedianow/df_tokopedianow/build.gradle b/dynamic-features/tokopedianow/df_tokopedianow/build.gradle
index 4232500d80d..b2f5d7c014f 100644
--- a/dynamic-features/tokopedianow/df_tokopedianow/build.gradle
+++ b/dynamic-features/tokopedianow/df_tokopedianow/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/travel/df_travel/build.gradle b/dynamic-features/travel/df_travel/build.gradle
index d2eebf8e610..0b6cd964fdc 100644
--- a/dynamic-features/travel/df_travel/build.gradle
+++ b/dynamic-features/travel/df_travel/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/travel/df_travel_lib/build.gradle b/dynamic-features/travel/df_travel_lib/build.gradle
index ec21958bea1..db5172d6f81 100644
--- a/dynamic-features/travel/df_travel_lib/build.gradle
+++ b/dynamic-features/travel/df_travel_lib/build.gradle
@@ -28,6 +28,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/user/df_user_liveness/build.gradle b/dynamic-features/user/df_user_liveness/build.gradle
index b964b85f01a..04f8aa53b35 100644
--- a/dynamic-features/user/df_user_liveness/build.gradle
+++ b/dynamic-features/user/df_user_liveness/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 repositories{
     flatDir {
@@ -34,6 +35,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/dynamic-features/user/df_user_settings/build.gradle b/dynamic-features/user/df_user_settings/build.gradle
index f1d5375c04f..a2921e814c5 100644
--- a/dynamic-features/user/df_user_settings/build.gradle
+++ b/dynamic-features/user/df_user_settings/build.gradle
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.dynamic-feature'
 apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.feature'
+apply plugin: 'robust'
 
 buildscript {
     repositories {
@@ -28,6 +29,12 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
 
+    buildTypes {
+        release {
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
+        }
+    }
+
     // Specifies 4 flavor dimensions.
     flavorDimensions "tier", "client"
 
diff --git a/features/communication/attachvoucher/build.gradle b/features/communication/attachvoucher/build.gradle
index 9dbcdc1dcc2..84cf831de91 100755
--- a/features/communication/attachvoucher/build.gradle
+++ b/features/communication/attachvoucher/build.gradle
@@ -21,11 +21,12 @@ android {
         viewBinding true
     }
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
+
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11.toString()
     }
 }
 
diff --git a/features/communication/inbox/build.gradle b/features/communication/inbox/build.gradle
index 5eb6202fbd9..65d0673e7f9 100644
--- a/features/communication/inbox/build.gradle
+++ b/features/communication/inbox/build.gradle
@@ -26,11 +26,11 @@ android {
         testInstrumentationRunner "com.tokopedia.test.application.environment.InstrumentationTestRunner"
     }
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 }
 
diff --git a/features/communication/orderhistory/build.gradle b/features/communication/orderhistory/build.gradle
index 9e8f48b99a5..bd07385e952 100644
--- a/features/communication/orderhistory/build.gradle
+++ b/features/communication/orderhistory/build.gradle
@@ -20,7 +20,7 @@ android {
         testInstrumentationRunner "com.tokopedia.test.application.environment.InstrumentationTestRunner"
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 
     buildFeatures {
diff --git a/features/communication/topchat/build.gradle b/features/communication/topchat/build.gradle
index 8a8309df307..8cb4a93b86b 100644
--- a/features/communication/topchat/build.gradle
+++ b/features/communication/topchat/build.gradle
@@ -31,11 +31,11 @@ android {
         testInstrumentationRunner "com.tokopedia.test.application.environment.InstrumentationTestRunner"
     }
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
     sourceSets {
         test {
diff --git a/features/content/feed/src/main/java/com/tokopedia/feedplus/view/fragment/FeedPlusContainerFragment.kt b/features/content/feed/src/main/java/com/tokopedia/feedplus/view/fragment/FeedPlusContainerFragment.kt
index 9e752b3db3e..7e36846f925 100644
--- a/features/content/feed/src/main/java/com/tokopedia/feedplus/view/fragment/FeedPlusContainerFragment.kt
+++ b/features/content/feed/src/main/java/com/tokopedia/feedplus/view/fragment/FeedPlusContainerFragment.kt
@@ -10,6 +10,7 @@ import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.annotation.Keep
 import androidx.appcompat.widget.Toolbar
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
@@ -89,6 +90,7 @@ import com.tokopedia.feedplus.view.customview.FeedFloatingButton
 private const val FEED_PAGE = "feed"
 private const val BROADCAST_VISIBLITY = "BROADCAST_VISIBILITY"
 
+@Keep
 class FeedPlusContainerFragment : BaseDaggerFragment(), FragmentListener, AllNotificationListener, FeedMainToolbar.OnToolBarClickListener,PostProgressUpdateView.PostUpdateSwipe, FeedPlusContainerListener {
 
     private var showOldToolbar: Boolean = false
diff --git a/features/content/feed_component/build.gradle b/features/content/feed_component/build.gradle
index 208396aaf1f..7abb220729a 100644
--- a/features/content/feed_component/build.gradle
+++ b/features/content/feed_component/build.gradle
@@ -23,7 +23,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/content/play/build.gradle b/features/content/play/build.gradle
index c30e358ecbc..267961a5a78 100644
--- a/features/content/play/build.gradle
+++ b/features/content/play/build.gradle
@@ -26,7 +26,7 @@ android {
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     defaultConfig {
diff --git a/features/content/play_broadcaster/build.gradle b/features/content/play_broadcaster/build.gradle
index 80010a5f2d3..b7b27ea675e 100644
--- a/features/content/play_broadcaster/build.gradle
+++ b/features/content/play_broadcaster/build.gradle
@@ -15,12 +15,12 @@ buildscript {
 
 android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     defaultConfig {
diff --git a/features/content/play_common/build.gradle b/features/content/play_common/build.gradle
index 3030a5a5a9c..a7be2bbcd23 100644
--- a/features/content/play_common/build.gradle
+++ b/features/content/play_common/build.gradle
@@ -4,12 +4,12 @@ apply plugin: 'io.hansel.preprocessor.module'
 
 android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     buildFeatures {
diff --git a/features/content/play_test/build.gradle b/features/content/play_test/build.gradle
index e0dcd768b01..c305e1ecaad 100644
--- a/features/content/play_test/build.gradle
+++ b/features/content/play_test/build.gradle
@@ -3,12 +3,12 @@ apply from: "$rootProject.projectDir/buildconfig/kotlin/module.gradle"
 
 android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = JavaVersion.VERSION_11
     }
 }
 
diff --git a/features/discovery/autocomplete/build.gradle b/features/discovery/autocomplete/build.gradle
index 954cbb13b68..3133de114d2 100644
--- a/features/discovery/autocomplete/build.gradle
+++ b/features/discovery/autocomplete/build.gradle
@@ -25,7 +25,7 @@ android {
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
diff --git a/features/discovery/autocomplete_component/build.gradle b/features/discovery/autocomplete_component/build.gradle
index 17bc8626788..a345aaabdd1 100644
--- a/features/discovery/autocomplete_component/build.gradle
+++ b/features/discovery/autocomplete_component/build.gradle
@@ -31,7 +31,7 @@ android {
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     buildFeatures {
diff --git a/features/discovery/discovery_common/build.gradle b/features/discovery/discovery_common/build.gradle
index 794ddb426e7..33bd49f4cbf 100644
--- a/features/discovery/discovery_common/build.gradle
+++ b/features/discovery/discovery_common/build.gradle
@@ -2,6 +2,23 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 apply plugin: 'kotlin-android-extensions'
 apply from: "$rootProject.projectDir/buildconfig/kotlin/module.gradle"
+apply plugin: 'io.hansel.preprocessor.module'
+
+repositories {
+    maven { url 'https://maven-hansel.tokopedia.com/maven' }
+}
+
+buildscript {
+    repositories {
+        google()
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
+        mavenCentral()
+        maven { url 'https://maven-hansel.tokopedia.com/maven' }
+    }
+    dependencies {
+        classpath rootProject.ext.miscDependencies.hansel
+    }
+}
 
 dependencies {
     implementation projectOrAar(rootProject.ext.features.resourcesCommon)
diff --git a/features/discovery/filter/build.gradle b/features/discovery/filter/build.gradle
index 03d7f6cae4a..a09528cd0d7 100644
--- a/features/discovery/filter/build.gradle
+++ b/features/discovery/filter/build.gradle
@@ -24,8 +24,8 @@ buildscript {
 }
 
 android {
-    viewBinding {
-        enabled true
+    buildFeatures {
+        viewBinding true
     }
 }
 
diff --git a/features/discovery/global_nav_widget/build.gradle b/features/discovery/global_nav_widget/build.gradle
index 2570b0bfa09..c88241fc4bf 100644
--- a/features/discovery/global_nav_widget/build.gradle
+++ b/features/discovery/global_nav_widget/build.gradle
@@ -5,9 +5,24 @@ apply from: "$rootProject.projectDir/buildconfig/module.gradle"
 apply plugin: 'io.hansel.preprocessor.module'
 
 android {
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
+    }
+    // For Kotlin projects
+    kotlinOptions {
+        jvmTarget = "11"
+    }
     buildFeatures {
         viewBinding true
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
+    }
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_11
+    }
 }
 repositories {
     maven { url 'https://maven-hansel.tokopedia.com/maven' }
diff --git a/features/discovery/home/build.gradle b/features/discovery/home/build.gradle
index 8672d37c327..0bbf1b9737b 100644
--- a/features/discovery/home/build.gradle
+++ b/features/discovery/home/build.gradle
@@ -29,7 +29,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     defaultConfig {
diff --git a/features/discovery/home_component/build.gradle b/features/discovery/home_component/build.gradle
index a840f500506..25ba499ae07 100644
--- a/features/discovery/home_component/build.gradle
+++ b/features/discovery/home_component/build.gradle
@@ -27,7 +27,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/discovery/home_nav/build.gradle b/features/discovery/home_nav/build.gradle
index 95e4c861c24..a24fa684c54 100644
--- a/features/discovery/home_nav/build.gradle
+++ b/features/discovery/home_nav/build.gradle
@@ -23,16 +23,16 @@ android{
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
 buildscript {
     repositories {
-        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
+        google()
         mavenCentral()
         maven { url 'https://maven-hansel.tokopedia.com/maven' }
-        google()
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
     }
     dependencies {
         classpath rootProject.ext.miscDependencies.hansel
diff --git a/features/discovery/recentview/build.gradle b/features/discovery/recentview/build.gradle
index 57c217f2bc8..58388faea17 100644
--- a/features/discovery/recentview/build.gradle
+++ b/features/discovery/recentview/build.gradle
@@ -23,7 +23,7 @@ buildscript {
 
 android {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/discovery/search/build.gradle b/features/discovery/search/build.gradle
index bfb11ca6e24..87f94269b1a 100644
--- a/features/discovery/search/build.gradle
+++ b/features/discovery/search/build.gradle
@@ -30,7 +30,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     defaultConfig {
diff --git a/features/entertainment/deals/build.gradle b/features/entertainment/deals/build.gradle
index 720e2d1b27d..84c73882647 100644
--- a/features/entertainment/deals/build.gradle
+++ b/features/entertainment/deals/build.gradle
@@ -31,11 +31,11 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 }
 
diff --git a/features/fintech/pdp_fintech_widget/build.gradle b/features/fintech/pdp_fintech_widget/build.gradle
index 87d11894b74..29c706d28f7 100644
--- a/features/fintech/pdp_fintech_widget/build.gradle
+++ b/features/fintech/pdp_fintech_widget/build.gradle
@@ -19,11 +19,11 @@ buildscript {
 
 android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8.toString()
+        jvmTarget = JavaVersion.VERSION_11.toString()
     }
 
     buildFeatures {
diff --git a/features/home/navigation-common/build.gradle b/features/home/navigation-common/build.gradle
index 22b55f65847..02e2002898c 100644
--- a/features/home/navigation-common/build.gradle
+++ b/features/home/navigation-common/build.gradle
@@ -9,8 +9,8 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 }
 
diff --git a/features/home/searchbar/build.gradle b/features/home/searchbar/build.gradle
index 5ebddf80c85..64871f599bb 100644
--- a/features/home/searchbar/build.gradle
+++ b/features/home/searchbar/build.gradle
@@ -30,12 +30,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
diff --git a/features/media/live-broadcaster/build.gradle b/features/media/live-broadcaster/build.gradle
index 04c9e471d61..43d636709fe 100644
--- a/features/media/live-broadcaster/build.gradle
+++ b/features/media/live-broadcaster/build.gradle
@@ -11,12 +11,12 @@ kapt {
 
 android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     buildTypes {
diff --git a/features/media/mediapicker-common/build.gradle b/features/media/mediapicker-common/build.gradle
index abd430eda8d..95ad892be0f 100644
--- a/features/media/mediapicker-common/build.gradle
+++ b/features/media/mediapicker-common/build.gradle
@@ -28,12 +28,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 
     testOptions {
diff --git a/features/media/mediapicker/build.gradle b/features/media/mediapicker/build.gradle
index 32d8e60cf52..e1ec22e7cac 100644
--- a/features/media/mediapicker/build.gradle
+++ b/features/media/mediapicker/build.gradle
@@ -33,12 +33,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = "11"
     }
 
     sourceSets {
diff --git a/features/merchant/brandlist/build.gradle b/features/merchant/brandlist/build.gradle
index d1db755690a..16dc9121495 100644
--- a/features/merchant/brandlist/build.gradle
+++ b/features/merchant/brandlist/build.gradle
@@ -24,7 +24,7 @@ android {
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
diff --git a/features/merchant/official_store/build.gradle b/features/merchant/official_store/build.gradle
index 428bf085bbe..b37d98633d8 100644
--- a/features/merchant/official_store/build.gradle
+++ b/features/merchant/official_store/build.gradle
@@ -26,7 +26,7 @@ android {
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
diff --git a/features/merchant/shop_showcase/src/main/java/com/tokopedia/shop_showcase/shop_showcase_tab/presentation/fragment/ShopPageShowcaseFragment.kt b/features/merchant/shop_showcase/src/main/java/com/tokopedia/shop_showcase/shop_showcase_tab/presentation/fragment/ShopPageShowcaseFragment.kt
index 07b0c776618..fcda3529147 100644
--- a/features/merchant/shop_showcase/src/main/java/com/tokopedia/shop_showcase/shop_showcase_tab/presentation/fragment/ShopPageShowcaseFragment.kt
+++ b/features/merchant/shop_showcase/src/main/java/com/tokopedia/shop_showcase/shop_showcase_tab/presentation/fragment/ShopPageShowcaseFragment.kt
@@ -7,6 +7,7 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.LinearLayout
+import androidx.annotation.Keep
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
 import androidx.recyclerview.widget.LinearLayoutManager
@@ -54,6 +55,7 @@ import javax.inject.Inject
 /**
  * Created by Rafli Syam on 05/03/2021
  */
+@Keep
 class ShopPageShowcaseFragment : BaseDaggerFragment(),
         HasComponent<ShopPageShowcaseComponent>,
     ShopPageFeaturedShowcaseListener,
diff --git a/features/ordermanagement/buyercancellationorder/build.gradle b/features/ordermanagement/buyercancellationorder/build.gradle
index 5a85fb81aca..0e714fd5843 100644
--- a/features/ordermanagement/buyercancellationorder/build.gradle
+++ b/features/ordermanagement/buyercancellationorder/build.gradle
@@ -22,7 +22,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = JavaVersion.VERSION_11
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/ordermanagement/snapshot/build.gradle b/features/ordermanagement/snapshot/build.gradle
index e24e447e444..5a9727077f5 100644
--- a/features/ordermanagement/snapshot/build.gradle
+++ b/features/ordermanagement/snapshot/build.gradle
@@ -22,7 +22,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/payment/gopay.kyc/build.gradle b/features/payment/gopay.kyc/build.gradle
index 31023fdc74c..8bc7a5b8376 100644
--- a/features/payment/gopay.kyc/build.gradle
+++ b/features/payment/gopay.kyc/build.gradle
@@ -19,12 +19,12 @@ buildscript {
 
 android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
     // For Kotlin projects
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/payment/pms/build.gradle b/features/payment/pms/build.gradle
index 14aa17fe96b..b727d6d13eb 100644
--- a/features/payment/pms/build.gradle
+++ b/features/payment/pms/build.gradle
@@ -19,12 +19,12 @@ buildscript {
 
 android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
     // For Kotlin projects
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/recharge/common_topup_bills/build.gradle b/features/recharge/common_topup_bills/build.gradle
index 98b52cae28e..f3f1160b022 100644
--- a/features/recharge/common_topup_bills/build.gradle
+++ b/features/recharge/common_topup_bills/build.gradle
@@ -19,11 +19,11 @@ buildscript {
 
 android {
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
     buildFeatures {
         viewBinding true
diff --git a/features/recharge/digital_checkout/build.gradle b/features/recharge/digital_checkout/build.gradle
index 0d37697f50a..ad4dbedd54c 100644
--- a/features/recharge/digital_checkout/build.gradle
+++ b/features/recharge/digital_checkout/build.gradle
@@ -34,11 +34,11 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 
     buildFeatures {
diff --git a/features/recharge/digital_product_detail/build.gradle b/features/recharge/digital_product_detail/build.gradle
index 487dcfe6e4d..d7191e9b1be 100644
--- a/features/recharge/digital_product_detail/build.gradle
+++ b/features/recharge/digital_product_detail/build.gradle
@@ -30,11 +30,11 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 }
 
diff --git a/features/recharge/recharge_component/build.gradle b/features/recharge/recharge_component/build.gradle
index 7b04292b954..9d240440018 100644
--- a/features/recharge/recharge_component/build.gradle
+++ b/features/recharge/recharge_component/build.gradle
@@ -27,7 +27,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/recharge/recharge_credit_card/build.gradle b/features/recharge/recharge_credit_card/build.gradle
index 0190275de0d..b1be792b2e6 100644
--- a/features/recharge/recharge_credit_card/build.gradle
+++ b/features/recharge/recharge_credit_card/build.gradle
@@ -29,12 +29,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 
     buildFeatures {
diff --git a/features/recharge/recharge_general/build.gradle b/features/recharge/recharge_general/build.gradle
index e120d2662cf..119b13fa6de 100644
--- a/features/recharge/recharge_general/build.gradle
+++ b/features/recharge/recharge_general/build.gradle
@@ -33,11 +33,11 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 }
 
diff --git a/features/recharge/rechargeocr/build.gradle b/features/recharge/rechargeocr/build.gradle
index c7bc9076694..22c5510f2b2 100644
--- a/features/recharge/rechargeocr/build.gradle
+++ b/features/recharge/rechargeocr/build.gradle
@@ -22,11 +22,11 @@ buildscript {
 
 android {
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 }
 
diff --git a/features/recharge/smart_bills/build.gradle b/features/recharge/smart_bills/build.gradle
index 79de6e96bd8..83571582755 100644
--- a/features/recharge/smart_bills/build.gradle
+++ b/features/recharge/smart_bills/build.gradle
@@ -27,11 +27,11 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
     buildFeatures {
         viewBinding true
diff --git a/features/transaction/buyerorder/build.gradle b/features/transaction/buyerorder/build.gradle
index e16fc8158d4..d00242b6e86 100644
--- a/features/transaction/buyerorder/build.gradle
+++ b/features/transaction/buyerorder/build.gradle
@@ -21,7 +21,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/transaction/checkout/build.gradle b/features/transaction/checkout/build.gradle
index e825235d6b5..a923ca49e16 100644
--- a/features/transaction/checkout/build.gradle
+++ b/features/transaction/checkout/build.gradle
@@ -25,7 +25,7 @@ android {
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     testOptions {
diff --git a/features/transaction/unifyorderhistory/build.gradle b/features/transaction/unifyorderhistory/build.gradle
index d77b5f6ba33..b66f98f95d4 100644
--- a/features/transaction/unifyorderhistory/build.gradle
+++ b/features/transaction/unifyorderhistory/build.gradle
@@ -29,7 +29,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/transaction/wishlist/build.gradle b/features/transaction/wishlist/build.gradle
index c6df68bbb52..6303d62e561 100644
--- a/features/transaction/wishlist/build.gradle
+++ b/features/transaction/wishlist/build.gradle
@@ -25,7 +25,7 @@ android {
         exclude 'META-INF/**'
     }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
     defaultConfig {
         multiDexEnabled true
diff --git a/features/transaction/wishlist_common/build.gradle b/features/transaction/wishlist_common/build.gradle
index 09dcae3cfe5..5b6099d4a35 100644
--- a/features/transaction/wishlist_common/build.gradle
+++ b/features/transaction/wishlist_common/build.gradle
@@ -24,8 +24,12 @@ android {
     packagingOptions {
         exclude 'META-INF/**'
     }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
+    }
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = JavaVersion.VERSION_11
     }
     buildFeatures {
         viewBinding true
diff --git a/features/travel/flight/build.gradle b/features/travel/flight/build.gradle
index bfcac4ec401..bccc20bd727 100644
--- a/features/travel/flight/build.gradle
+++ b/features/travel/flight/build.gradle
@@ -29,12 +29,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 
     buildFeatures{
diff --git a/features/travel/hotel/build.gradle b/features/travel/hotel/build.gradle
index d1ff0743eb1..c79493f799a 100644
--- a/features/travel/hotel/build.gradle
+++ b/features/travel/hotel/build.gradle
@@ -39,11 +39,11 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 
     buildFeatures{
diff --git a/features/user/logout/src/main/java/com/tokopedia/logout/view/LogoutActivity.kt b/features/user/logout/src/main/java/com/tokopedia/logout/view/LogoutActivity.kt
index 9da496896f8..467207e5bb9 100644
--- a/features/user/logout/src/main/java/com/tokopedia/logout/view/LogoutActivity.kt
+++ b/features/user/logout/src/main/java/com/tokopedia/logout/view/LogoutActivity.kt
@@ -99,7 +99,7 @@ class LogoutActivity : BaseSimpleActivity(), HasComponent<LogoutComponent> {
         component.inject(this)
         userSession = UserSession(this)
 
-        getParams()
+        getParam()
 
         initTetraDebugger()
         initObservable()
@@ -115,7 +115,7 @@ class LogoutActivity : BaseSimpleActivity(), HasComponent<LogoutComponent> {
         }
     }
 
-    private fun getParams() {
+    private fun getParam() {
         if (intent.extras != null) {
             isReturnToHome = intent.extras?.getBoolean(ApplinkConstInternalUserPlatform.PARAM_IS_RETURN_HOME, true) as Boolean
             isClearDataOnly = intent.extras?.getBoolean(ApplinkConstInternalUserPlatform.PARAM_IS_CLEAR_DATA_ONLY, false) as Boolean
diff --git a/features/user/settingnotif/build.gradle b/features/user/settingnotif/build.gradle
index cf5d8f4b250..ab9427d7a7d 100644
--- a/features/user/settingnotif/build.gradle
+++ b/features/user/settingnotif/build.gradle
@@ -8,12 +8,12 @@ repositories {
 
 android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
diff --git a/features/user/updateinactivephone/build.gradle b/features/user/updateinactivephone/build.gradle
index 1363c1b4545..47e357abe0c 100644
--- a/features/user/updateinactivephone/build.gradle
+++ b/features/user/updateinactivephone/build.gradle
@@ -22,12 +22,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        jvmTarget = JavaVersion.VERSION_11
     }
 
     buildFeatures {
diff --git a/features/user/usercomponents/build.gradle b/features/user/usercomponents/build.gradle
index 876c98db66f..5b2e7dfaf30 100644
--- a/features/user/usercomponents/build.gradle
+++ b/features/user/usercomponents/build.gradle
@@ -22,6 +22,15 @@ android {
         testInstrumentationRunner "com.tokopedia.test.application.environment.InstrumentationTestRunner"
     }
 
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
+    }
+
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_11
+    }
+
     buildFeatures {
         viewBinding true
     }
diff --git a/gradle.properties b/gradle.properties
index 55d51bed8c1..9668b2de44a 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -25,7 +25,6 @@ authToken=jp_sv3sov9akgpn22fa8m97u7milt
 enableStability=false
 android.useAndroidX=true
 android.enableJetifier=true
-android.enableR8=false
 android.defaults.buildfeatures.buildconfig=false
 android.defaults.buildfeatures.aidl=false
 android.defaults.buildfeatures.renderscript=false
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
index 7d969272f8f..7546beafdf6 100644
--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.9-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-bin.zip
diff --git a/libraries/akamai_bot_lib/build.gradle b/libraries/akamai_bot_lib/build.gradle
index f7610db7594..7c9c1d5977e 100644
--- a/libraries/akamai_bot_lib/build.gradle
+++ b/libraries/akamai_bot_lib/build.gradle
@@ -13,6 +13,16 @@ buildscript {
     }
 }
 
+android {
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
+    }
+    kotlinOptions {
+        jvmTarget = '11'
+    }
+}
+
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation rootProject.ext.tkpdInternalLibDependencies.globalconfig
diff --git a/libraries/analytics_debugger/build.gradle b/libraries/analytics_debugger/build.gradle
index 8fea729364b..b280f883e35 100644
--- a/libraries/analytics_debugger/build.gradle
+++ b/libraries/analytics_debugger/build.gradle
@@ -14,7 +14,7 @@ buildscript {
 }
 android {
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8.toString()
+        jvmTarget = JavaVersion.VERSION_11.toString()
     }
     buildFeatures {
         viewBinding true
diff --git a/libraries/analytics_debugger_noop/build.gradle b/libraries/analytics_debugger_noop/build.gradle
index af374bfd8a8..b46d1f9dc5a 100644
--- a/libraries/analytics_debugger_noop/build.gradle
+++ b/libraries/analytics_debugger_noop/build.gradle
@@ -8,8 +8,8 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 }
 
diff --git a/libraries/applink/build.gradle b/libraries/applink/build.gradle
index d02aa1a056d..89401d75e60 100644
--- a/libraries/applink/build.gradle
+++ b/libraries/applink/build.gradle
@@ -15,12 +15,12 @@ buildscript {
 
 android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8.toString()
+        jvmTarget = JavaVersion.VERSION_11.toString()
     }
 }
 
diff --git a/libraries/composite_lib/analytic_param/build.gradle b/libraries/composite_lib/analytic_param/build.gradle
index ae2d7cc2eec..8fb97c7029b 100644
--- a/libraries/composite_lib/analytic_param/build.gradle
+++ b/libraries/composite_lib/analytic_param/build.gradle
@@ -7,8 +7,8 @@ dependencies {
     implementation rootProject.ext.supportLibDependencies.kotlinSupportJdk8
 }
 
-sourceCompatibility = "7"
-targetCompatibility = "7"
+sourceCompatibility = "11"
+targetCompatibility = "11"
 buildscript {
     repositories {
         google()
@@ -23,12 +23,12 @@ repositories {
 }
 compileKotlin {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 compileTestKotlin {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
diff --git a/libraries/composite_lib/analytic_testapp/build.gradle b/libraries/composite_lib/analytic_testapp/build.gradle
index 10a2eb3ba0b..ff86be0af6d 100644
--- a/libraries/composite_lib/analytic_testapp/build.gradle
+++ b/libraries/composite_lib/analytic_testapp/build.gradle
@@ -24,8 +24,8 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility 1.8
-        targetCompatibility 1.8
+        sourceCompatibility 11
+        targetCompatibility 11
     }
 }
 
diff --git a/libraries/composite_lib/annotation_processor_analytic/build.gradle b/libraries/composite_lib/annotation_processor_analytic/build.gradle
index 73561916612..8ada8cf9d4b 100644
--- a/libraries/composite_lib/annotation_processor_analytic/build.gradle
+++ b/libraries/composite_lib/annotation_processor_analytic/build.gradle
@@ -6,8 +6,8 @@ dependencies {
     implementation rootProject.ext.supportLibDependencies.kotlinSupportJdk8
 }
 
-sourceCompatibility = "8"
-targetCompatibility = "8"
+sourceCompatibility = "11"
+targetCompatibility = "11"
 
 buildscript {
     repositories {
@@ -21,12 +21,12 @@ buildscript {
 
 compileKotlin {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 compileTestKotlin {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
diff --git a/libraries/composite_lib/build.gradle b/libraries/composite_lib/build.gradle
index e207d9cb952..b42b42ccf7a 100644
--- a/libraries/composite_lib/build.gradle
+++ b/libraries/composite_lib/build.gradle
@@ -24,6 +24,7 @@ buildscript {
                 username = "${artifactory_username}"
                 password = "${artifactory_password}"
             }
+            allowInsecureProtocol = true
         }
     }
     dependencies {
diff --git a/libraries/composite_lib/buildSrc/build.gradle b/libraries/composite_lib/buildSrc/build.gradle
index 55cd5122c34..f06e7dff597 100644
--- a/libraries/composite_lib/buildSrc/build.gradle
+++ b/libraries/composite_lib/buildSrc/build.gradle
@@ -32,8 +32,8 @@ task createClasspathManifest {
     }
 }
 
-compileKotlin { kotlinOptions { jvmTarget = "1.8" } }
-compileTestKotlin { kotlinOptions { jvmTarget = "1.8" } }
+compileKotlin { kotlinOptions { jvmTarget = "11" } }
+compileTestKotlin { kotlinOptions { jvmTarget = "11" } }
 
 dependencies {
     testRuntimeOnly files(createClasspathManifest)
diff --git a/libraries/composite_lib/gqlquery_annotation/build.gradle b/libraries/composite_lib/gqlquery_annotation/build.gradle
index 674677a9aa3..094c64baf02 100644
--- a/libraries/composite_lib/gqlquery_annotation/build.gradle
+++ b/libraries/composite_lib/gqlquery_annotation/build.gradle
@@ -18,17 +18,17 @@ buildscript {
 
 compileKotlin {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 compileTestKotlin {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
-sourceCompatibility = "8"
-targetCompatibility = "8"
+sourceCompatibility = "11"
+targetCompatibility = "11"
 
 project.ext {
     artifactId = 'gqlquery_annotation'
diff --git a/libraries/composite_lib/gqlquery_annotation_processor/build.gradle b/libraries/composite_lib/gqlquery_annotation_processor/build.gradle
index fada56be460..2a9c2739231 100644
--- a/libraries/composite_lib/gqlquery_annotation_processor/build.gradle
+++ b/libraries/composite_lib/gqlquery_annotation_processor/build.gradle
@@ -20,17 +20,17 @@ dependencies {
     kapt rootProject.ext.miscDependencies.autoService
 }
 
-sourceCompatibility = "8"
-targetCompatibility = "8"
+sourceCompatibility = "11"
+targetCompatibility = "11"
 
 compileKotlin {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 compileTestKotlin {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 }
 
diff --git a/libraries/composite_lib/settings.gradle b/libraries/composite_lib/settings.gradle
index 54be78ca0d6..1d853b73ae0 100644
--- a/libraries/composite_lib/settings.gradle
+++ b/libraries/composite_lib/settings.gradle
@@ -8,6 +8,7 @@ buildCache {
     remote(HttpBuildCache) {
         url = "${ci_url}"
         push = isCiServer
+        allowInsecureProtocol = true
 //        allowUntrustedServer = true
         credentials {
             username = "${ci_username}"
diff --git a/libraries/device_info/build.gradle b/libraries/device_info/build.gradle
index 63687d04d0c..00de3a93aa2 100644
--- a/libraries/device_info/build.gradle
+++ b/libraries/device_info/build.gradle
@@ -18,12 +18,12 @@ android {
         multiDexEnabled true
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8.toString()
+        jvmTarget = JavaVersion.VERSION_11.toString()
     }
 }
 
diff --git a/libraries/graphql/src/main/java/com/tokopedia/graphql/GraphqlConstant.java b/libraries/graphql/src/main/java/com/tokopedia/graphql/GraphqlConstant.java
index fff77f00c3a..4292887528f 100644
--- a/libraries/graphql/src/main/java/com/tokopedia/graphql/GraphqlConstant.java
+++ b/libraries/graphql/src/main/java/com/tokopedia/graphql/GraphqlConstant.java
@@ -14,6 +14,7 @@ public interface GraphqlConstant {
         String QUERY = "query";
         String VARIABLES = "variables";
         String OPERATION_NAME = "operationName";
+        String MD5 = "md5";
         String DATA = "data";
         String ERROR = "errors";
     }
diff --git a/libraries/graphql/src/main/java/com/tokopedia/graphql/data/model/GraphqlRequest.kt b/libraries/graphql/src/main/java/com/tokopedia/graphql/data/model/GraphqlRequest.kt
index c96622d473d..0dd9ee6159c 100644
--- a/libraries/graphql/src/main/java/com/tokopedia/graphql/data/model/GraphqlRequest.kt
+++ b/libraries/graphql/src/main/java/com/tokopedia/graphql/data/model/GraphqlRequest.kt
@@ -52,7 +52,7 @@ class GraphqlRequest {
     @Transient
     var isNoCache = false
 
-    @Expose(serialize = false, deserialize = false)
+    @SerializedName(GraphqlConstant.GqlApiKeys.MD5)
     var md5: String
         private set
 
diff --git a/libraries/inappupdate/build.gradle b/libraries/inappupdate/build.gradle
index 52bdee2ecd2..1d034439b0e 100644
--- a/libraries/inappupdate/build.gradle
+++ b/libraries/inappupdate/build.gradle
@@ -13,6 +13,16 @@ buildscript {
     }
 }
 
+android {
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
+    }
+    kotlinOptions {
+        jvmTarget = '11'
+    }
+}
+
 dependencies {
     implementation rootProject.ext.playServicesDependencies.core
     implementation projectOrAar(rootProject.ext.libraries.logger)
diff --git a/libraries/logger/build.gradle b/libraries/logger/build.gradle
index 1a7dc6c52ac..34cf147c0ea 100644
--- a/libraries/logger/build.gradle
+++ b/libraries/logger/build.gradle
@@ -34,6 +34,16 @@ dependencies {
     testImplementation rootProject.ext.testDependencies.kotlinMockito
 }
 
+android {
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
+    }
+    kotlinOptions {
+        jvmTarget = '11'
+    }
+}
+
 project.ext {
     artifactId = 'logger'
     groupId = 'com.tokopedia.library'
diff --git a/libraries/macrobenchmark/build.gradle b/libraries/macrobenchmark/build.gradle
index 9f2f02e96b6..c786b2dcd12 100644
--- a/libraries/macrobenchmark/build.gradle
+++ b/libraries/macrobenchmark/build.gradle
@@ -17,11 +17,11 @@ android {
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
     kotlinOptions {
-        jvmTarget = '1.8'
+        jvmTarget = '11'
     }
 
     signingConfigs {
diff --git a/libraries/remote_config/src/main/res/xml/remote_config_default.xml b/libraries/remote_config/src/main/res/xml/remote_config_default.xml
index 3862aa32327..9720920bc8c 100644
--- a/libraries/remote_config/src/main/res/xml/remote_config_default.xml
+++ b/libraries/remote_config/src/main/res/xml/remote_config_default.xml
@@ -646,4 +646,8 @@
         <key>android_review_form_new_enable</key>
         <value>true</value>
     </entry>
+    <entry>
+        <key>android_robust_tester</key>
+        <value></value>
+    </entry>
 </defaultsMap>
diff --git a/libraries/tokopatch/build.gradle b/libraries/tokopatch/build.gradle
index 59135ab9363..be0d5fa5aae 100644
--- a/libraries/tokopatch/build.gradle
+++ b/libraries/tokopatch/build.gradle
@@ -19,12 +19,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     buildTypes {
diff --git a/libraries/track/build.gradle b/libraries/track/build.gradle
index a0427994478..ec495183568 100644
--- a/libraries/track/build.gradle
+++ b/libraries/track/build.gradle
@@ -23,4 +23,15 @@ project.ext {
 
 apply from: '../../publish_local.gradle'
 apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-android-extensions'
\ No newline at end of file
+apply plugin: 'kotlin-android-extensions'
+
+android {
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
+    }
+
+    kotlinOptions {
+        jvmTarget = "11"
+    }
+}
\ No newline at end of file
diff --git a/libraries/utils/build.gradle b/libraries/utils/build.gradle
index 7051bfc4bf2..d650b36fd08 100644
--- a/libraries/utils/build.gradle
+++ b/libraries/utils/build.gradle
@@ -40,8 +40,11 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
+    }
+    kotlinOptions {
+        jvmTarget = '11'
     }
 }
 
diff --git a/sellerapp/build.gradle b/sellerapp/build.gradle
index 09354eb6fe6..cf1a1be4ea8 100644
--- a/sellerapp/build.gradle
+++ b/sellerapp/build.gradle
@@ -1,7 +1,4 @@
 apply plugin: 'com.android.application'
-if (project.hasProperty("auto-patch")) {
-    apply plugin: 'auto-patch-plugin'
-}
 apply plugin: 'com.google.firebase.crashlytics'
 apply plugin: 'robust'
 apply plugin: 'io.hansel.preprocessor'
@@ -103,7 +100,7 @@ android {
             if (devMode()) {
                 signingConfig signingConfigs.debug
             }
-            debuggable = project.hasProperty('enableDebug') && enableDebug == "true"
+            debuggable = rootProject.hasProperty('enableDebug') && rootProject.property("enableDebug") == "true"
             ext.enableHansel = !project.hasProperty('hanselDisableRelease')
             minifyEnabled true
             zipAlignEnabled true
@@ -138,8 +135,8 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 
     configurations.all {
diff --git a/sellerapp/proguard-rules.txt b/sellerapp/proguard-rules.txt
index 03dc388a011..07849239a39 100644
--- a/sellerapp/proguard-rules.txt
+++ b/sellerapp/proguard-rules.txt
@@ -1,5 +1,6 @@
-
-
+-printusage usage.txt
+-printseeds seeds.txt
+-printconfiguration config.txt
 
 # Obfuscation parameters:
 -dontwarn java.nio.file.Files
@@ -41,11 +42,12 @@
 -keep public class com.google.android.gms.common.internal.safeparcel.SafeParcelable {
     public static final *** NULL;
 }
--keepnames @com.google.android.gms.common.annotation.KeepName class *
+-keep @com.google.android.gms.common.annotation.KeepName class *
 -keepclassmembernames class * {
     @com.google.android.gms.common.annotation.KeepName *;
 }
--keepnames class * implements android.os.Parcelable {
+
+-keep class * implements android.os.Parcelable {
     public static final ** CREATOR;
 }
 #end com.google.android.gms
@@ -165,8 +167,8 @@
 -dontwarn kotlinx.coroutines.flow.**
 -dontwarn kotlinx.coroutines.reactive.**
 
--keepnames class kotlinx.coroutines.internal.MainDispatcherFactory {}
--keepnames class kotlinx.coroutines.CoroutineExceptionHandler {}
+-keep class kotlinx.coroutines.internal.MainDispatcherFactory {}
+-keep class kotlinx.coroutines.CoroutineExceptionHandler {}
 
 # -- App indexing
 -dontwarn com.google.firebase.appindexing.**
@@ -287,9 +289,9 @@
 -dontwarn org.bouncycastle.**
 
 # Keep fragment class name
--keepnames class com.tokopedia.dynamicfeatures.DFInstallerFragment
--keepnames class com.tokopedia.review.feature.reading.presentation.fragment.ShopReviewFragment
--keepnames class com.tokopedia.shop_showcase.shop_showcase_tab.presentation.fragment.ShopPageShowcaseFragment
+-keep class com.tokopedia.dynamicfeatures.DFInstallerFragment
+-keep class com.tokopedia.review.feature.reading.presentation.fragment.ShopReviewFragment
+-keep class com.tokopedia.shop_showcase.shop_showcase_tab.presentation.fragment.ShopPageShowcaseFragment
 
 # feed_shop
 -keepnames class com.tokopedia.feed_shop.shop.view.fragment.FeedShopFragment
@@ -298,4 +300,12 @@
 -keep class com.tokopedia.stability.**{*;}
 -keepclassmembers class **{
 public static com.tokopedia.stability.ChangeDelegate *;
-}
\ No newline at end of file
+}
+
+-keepclassmembers,allowobfuscation class * {
+ @com.google.gson.annotations.SerializedName <fields>;
+}
+
+-keep class com.google.gson.reflect.TypeToken { *; }
+-keep class * extends com.google.gson.reflect.TypeToken
+-keep class kotlin.coroutines.Continuation
\ No newline at end of file
diff --git a/settings.gradle b/settings.gradle
index 8accddbc235..abc39256151 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -10,11 +10,13 @@ buildCache {
     remote(HttpBuildCache) {
         url = "${ci_url}"
         push = isCiServer
+        allowInsecureProtocol = true
         enabled = enableRemoteCache
         credentials {
             username = "${ci_username}"
             password = "${ci_password}"
         }
+        allowInsecureProtocol = true
     }
 }
 
diff --git a/testapp/build.gradle b/testapp/build.gradle
index b73b5ecf5f6..0bf388750cc 100644
--- a/testapp/build.gradle
+++ b/testapp/build.gradle
@@ -68,8 +68,8 @@ android {
         buildConfig true
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 }
 dependencies {
diff --git a/testing/joint/dg_transaction/build.gradle b/testing/joint/dg_transaction/build.gradle
index 935538b6682..e05926d31e6 100644
--- a/testing/joint/dg_transaction/build.gradle
+++ b/testing/joint/dg_transaction/build.gradle
@@ -11,7 +11,7 @@ android {
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     testOptions {
diff --git a/testing/joint/inactivephone_otp/build.gradle b/testing/joint/inactivephone_otp/build.gradle
index 733bf291bb0..00a4bfa821f 100644
--- a/testing/joint/inactivephone_otp/build.gradle
+++ b/testing/joint/inactivephone_otp/build.gradle
@@ -10,7 +10,7 @@ android {
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     testOptions {
diff --git a/testing/joint/pdp_checkout/build.gradle b/testing/joint/pdp_checkout/build.gradle
index 6e50acddfc3..e48160514d8 100644
--- a/testing/joint/pdp_checkout/build.gradle
+++ b/testing/joint/pdp_checkout/build.gradle
@@ -10,7 +10,7 @@ android {
     }
 
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
     }
 
     testOptions {
diff --git a/tools/dynamic_feature_inclusion.gradle b/tools/dynamic_feature_inclusion.gradle
index b7fae413c06..728561df503 100644
--- a/tools/dynamic_feature_inclusion.gradle
+++ b/tools/dynamic_feature_inclusion.gradle
@@ -35,7 +35,7 @@ ext {
         }
     }
 
-    generateAssetFile = {Project project, LinkedHashSet<String> strings ->
+    generateAssetFile = {Project project, strings ->
         def projectName = project.name
         def assetDir = new File(projectName + "/src/main/assets")
         def fileAsset = new File(projectName + "/src/main/assets/df.cfg")
